// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//         http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";
package food;


service FoodService {
    rpc GetVendors (SupplierRequest) returns (SupplierReply) {}
    rpc GetIngredientInfo (VendorRequest) returns (VendorReply) {}
}

message SupplierRequest {
    string ingredient = 1;
}

message SupplierReply {
    string vendors = 1;
}

message VendorRequest {
    string ingredient = 1;
    string vendorName = 2;
}

message VendorReply {
    int32 inventoryCount = 1;
    float price = 2;
}
